apps/api/jwt_auth/views.py

```diff
@@ -0,0 +1,36 @@
+from rest_framework import exceptions
+from rest_framework.permissions import IsAuthenticated
+from rest_framework.views import APIView
+from rest_framework_jwt.authentication import JSONWebTokenAuthentication
+from rest_framework_jwt.compat import get_user_model
+
+from apps.api.models import APIKey
+
+
+class JWTKeyAuthentication(JSONWebTokenAuthentication):
+
+    def authenticate_credentials(self, payload):
+        """
+        Returns an active user that matches the payload's user id and email.
+        """
+        User = get_user_model()
+        try:
+            api_key = APIKey.objects.get(key=payload['iss'])
+        except APIKey.DoesNotExist:
+            msg = 'Invalid API Key.'
+            raise exceptions.AuthenticationFailed(msg)
+
+        user = api_key.user
```

Comment:
I wouldn't personally assign `user` here, it's just one access away and isn't used much. Doesn't matter much just looked like it might be leftover from a refactoring.